tools/caml_tex.ml

```diff
@@ -132,12 +132,10 @@ module Toplevel = struct
     if startchar >= 0 then
       locs := (startchar, endchar) :: !locs
 
-  (** Record locations in the main error and suberrors without printing them *)
+  (** Record the main location instead of printing it *)
   let printer_register_locs =
     { Location.batch_mode_printer with
-      pp_main_loc = (fun _ _ _ loc -> register_loc loc);
-      pp_submsg_loc = (fun _ _ _ loc -> register_loc loc);
-    }
+      pp_main_loc = (fun _ _ _ loc -> register_loc loc) }
```

Comment:
I imagine that it was easier to revert this fix?

Reply:
Ah, sorry, I'll amend to un-revert this part. Note: the commit message for a241675676a742053c6e785f017889f253d060b3 says "List printer fields explicitly to avoid the same problem from happening again", but the commit doesn't actually do it (if I understand correctly it should not use `with` at all), do you think I should do it in passing?

Reply:
If you have the time it would be nice, in particular if there is clear separation between the updated and copied fields as I suggested in https://github.com/ocaml/ocaml/pull/2307#discussion_r264225541 . But the current compromise is perfectly fine with me.

Reply:
Thanks for the link, I had totally forgotten that context. Will do.

Reply:
I updated the PR accordingly, what do you think?

Reply:
It looks good, thanks!